src/approve.ts

```diff
@@ -31,6 +31,7 @@ export const constructApproveToken = (
     tokenAddress,
     overrides = {}
   ) => {
+    // @TODO consider caching `spender`
```

Comment:
good idea, gotta do it per network

Reply:
if cached inside `construct*` closure, no need to care about network